NomNomAPI/Controllers/FoodController.cs

```diff
@@ -108,5 +108,39 @@ public async Task<ActionResult<List<FoodItem>>> GetFoodItemsByPriceRange([FromQu
 
             return Ok(foodItems);
         }
+
+        [HttpGet]
+        public async Task<IActionResult> GetFoodItems(
+            [FromQuery] int? storeId = null,
+            [FromQuery] string? category = null,
+            [FromQuery] DateTime? expirationDate = null,
+            [FromQuery] double? minPrice = null,
+            [FromQuery] double? maxPrice = null,
+            [FromQuery] double? minDiscount = null,
+            [FromQuery] string? name = null,
+            [FromQuery] bool? isVegan = null,
+            [FromQuery] string? description = null)  // Add this parameter
+        {
+            var foodItems = await _foodItemService.GetFoodItemsAsync(
+                storeId, category, expirationDate, minPrice, maxPrice, minDiscount, name, isVegan, description);
+
+            var result = foodItems.Select(f => new
+            {
+                f.Id,
+                f.StoreId,
+                f.Name,
+                f.Description,
+                f.Category,
+                OriginalPrice = f.Price,
+                DiscountedPrice = f.DiscountedPrice,
+                DiscountPercentage = (f.Price - f.DiscountedPrice) / f.Price * 100,
+                f.ExpirationDate,
+                f.ImageUrl,
+                f.IsVegan
+            });
```

Comment:
It is usually good to have model conversions in a separate static class to avoid code repetition (DRY) and keep controller level more clean. It also makes controllers easier to test since conversion logic can be tested separately